src/settings/JobProfiles/JobProfileDetailsRoute/JobProfileDetailsRoute.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  get,
+  find,
+} from 'lodash';
+
+import { JobProfileDetails } from '../JobProfileDetails';
+
+const JobProfileDetailsRoute = ({
+  resources: {
+    jobProfile,
+    mappingProfile,
+  },
+  history,
+  location,
+  match,
+}) => {
+  // `find` is used to make sure that the job profile that matches the id in url is displayed
+  const jobProfileRecord = find([get(jobProfile, 'records.0', {})], { id: match.params.id });
+  const mappingProfileRecord = find([get(mappingProfile, 'records.0', {})], { id: jobProfileRecord?.mappingProfileId });
```

Comment:
Please, could you clarify the next: in line 55 only 1 mapping profile is requrested by id, so why do we need to use find in this place?

Reply:
The same as for the `jobProfile`. It can be the flickering when the previous mapping profile is going to show for some time. Technically for mapping profile the find can be removed as `loading` anyways will be false as `jobProfile` won't be found because of the `find` in place.

Reply:
I think It can confuse someone without knowing the context. So, please, extend your comment or modify the code.

Reply:
Agree, I will check whether I can simplify and if not will extend the comment.